### ISSUE TITLE
[Addition] [Low Priority] Add more unit tests for the fixed search space implementation

### ISSUE THREAD

**Issue by ArlindKadra** [automl/Auto-PyTorch#179]
Add unit tests where we restrict the search space as follows:

- Skip Connections are disabled.
- Snapshot Ensembling is disabled.
- Weight Decay is disabled.
- Dropout is disabled.

Furthermore

-  Restrictions for the network head.

All of the above were given errors and are now patched.
